### PR TITLE
issue#203 - increase runtime file descriptors allowed by etcd from 1024 to 128000 - DO NOT MERGE YET

### PR DESCRIPTION
**Pull request by notjames**
nt

### CODE REVIEW

**Comment by notjames** [comment]
possible fix for https://github.com/samsung-cnct/kraken/issues/203

File: ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.kraken-etcd-ssl.part.jinja2 [hunk 1]
```diff
@@ -7,6 +7,7 @@
     Requires=kraken-networking.service
     After=kraken-networking.service
     [Service]
+    LimitNOFILE=128000
```
Comment (joejulian) [inline ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.kraken-etcd-ssl.part.jinja2]: Not needed here. This service only runs once to generate the needed certs.

**Comment by joejulian** [comment]
@coffeepac Per our discussion, I'm for merging this even without proving that it fixes the problem of running out of FDs. The etcd nodes only need a very few FDs for their other operations. Let's go ahead and give etcd and etcdEvents a huge chunk of them in case they're ever needed in the future.

**Comment by alejandroEsc** [comment]
@joejulian i have reservations about using bandaids like this. Can we please open an issue to track this value change here, and look at it later as we resolve this issue. I think this will cause other issues later on and loosing track of this would mean a hugely annoying bug later on.

**Comment by coffeepac** [comment]
I am unlikely to approve this.  the issue seems to be specific to etcd 3.2.5 and kubernetes 1.8.1.  the normal steady state for fd usage is ~80.  If we running at 10x that I would like to have something fail sooner then later.

**Comment by joejulian** [comment]
I have no skin in this one way or the other. Playing the devil's advocate, however, as long as we have sufficient monitoring setting this to a high number gives an opportunity to correct the problem before operations are affected should this happen in a production environment.

File: ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2 [hunk 2]
```diff
@@ -14,6 +14,7 @@
     After=kraken-networking.service
     Requires=kraken-networking.service
     [Service]
+    LimitNOFILE=128000
```
[the remark this answers is not in the document]
Comment (davidewatson) [reply ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2]: Also, it appears that CoreOS has already implemented a protective measure by reserving some file descriptors for internal use:

https://github.com/coreos/etcd/pull/3219

https://github.com/coreos/etcd/blob/bcef78c665c7c4bf2094ce2e45d2822cee7cd626/embed/etcd.go#L62

[the remark this answers is not in the document]
Comment (notjames) [reply ansible/roles/kraken.nodePool/kraken.nodePool.etcd/templates/etcd.units.etcd.part.jinja2]: I agree with not pushing this yet to prod. I appreciate everybody's input.

**Comment by joejulian** [comment]
The consensus is that we're not going to do this. I'm going to go ahead and close this PR. Feel free to reopen it if you think circumstances warrant it.

**Pull request closed without merging**